AppServer/google/appengine/tools/appengine_rpc.py

```diff
@@ -611,7 +617,8 @@ def Send(self, request_path, payload="",
           else:
             raise
     finally:
-      socket.setdefaulttimeout(old_timeout)
+      if not opener_has_timeout:
+        socket.setdefaulttimeout(old_timeout)
```

Comment:
@whoarethebritons , it couldn't, because it should be initialized by original defaulttimeout which is rewrote in the beginning of this method